### PR TITLE
Make all PHP files strict

### PR DESCRIPTION
**Pull request by oliverklee**
This make sure that the required types are indeed provided.

### CODE REVIEW

**Comment by Danon** [comment]
Hello, thanks for another PR!

We've introduced formal rules before into T-Regx libraries, and it was to use Fully Qualified Names: https://www.php.net/manual/en/language.namespaces.rules.php. But we didn't just *say*: "let's use it", but added "fqn" composer checker, that would validate that they're used.

This `declare(strict_types=1)` is a good idea, I can go with it, but it looks like something that can get forgotten on certain files. So if I were to add this, I would require a tool that can scan my project and validate that it exists everywhere.

I will probably see it as a feature update to my fork of: https://github.com/T-Regx/fqn-check. This is the tool that parses files, and validates FQNs. It would be easy to edit it so it also checks `declare()` types. Or perhaps there's already another tool that checks it?

**tl;dr;** I will merge this PR, when I have a composer script that can validate it.

PS: Thank you very much for your contributions to the lib!

**Comment by oliverklee** [comment]
Gentle ping @Danon for a review. :cookie:

**Comment by oliverklee** [comment]
@Danon Do you still need anything from me so you can merge this PR? (I'd like to reduce the number of open PRs as I rebase all of them after one has been merged, and some PRs also depend on other PRs.)
